internal/record/recording_https_proxy.go

```diff
@@ -178,21 +168,49 @@ func (r *RecordingHTTPSProxy) proxyRequest(w http.ResponseWriter, req *http.Requ
 	return resp, respBodyBytes, nil
 }
 
-func (r *RecordingHTTPSProxy) recordResponse(resp *http.Response, fileName string, body []byte) error {
+func (r *RecordingHTTPSProxy) recordResponse(recReq *store.RecordedRequest, resp *http.Response, fileName string, shaSum string, body []byte) error {
 	recordedResponse, err := store.NewRecordedResponse(resp, body)
 	if err != nil {
 		return err
 	}
+	recordPath := filepath.Join(r.recordingDir, fileName+".http.log")
 
-	recordedResponse.Body = r.redactor.Bytes(recordedResponse.Body)
+	var fileMode int
+	if fileName == shaSum {
+		// This is a single-request file, so overwrite it completely.
+		fileMode = os.O_TRUNC
+	} else {
+		// This file groups the requests by test-name, so append to it.
```

Comment:
What's the story for re-running a test (e.g I changed the code and I want to re-record).
With current code it'll end up appending to the existing file and the errors may be confusing?